### PR TITLE
Fix memory leak in relation editor widget

### PR DESCRIPTION
**Pull request by domi4484**
Fix memory leak in relation editor widget

### CODE REVIEW

File: src/gui/qgsrelationeditorwidget.h [hunk 1]
```diff
@@ -255,7 +255,7 @@ class GUI_EXPORT QgsRelationEditorWidget : public QgsAbstractRelationEditorWidge
     QTreeWidget *mMultiEditTreeWidget = nullptr;
     QObjectUniquePtr<QgsMapToolDigitizeFeature> mMapToolDigitize;
     QButtonGroup *mViewModeButtonGroup = nullptr;
-    QgsVectorLayerSelectionManager *mFeatureSelectionMgr = nullptr;
+    QSharedPointer<QgsVectorLayerSelectionManager>mFeatureSelectionMgr = nullptr;
```
Comment (nyalldawson) [inline src/gui/qgsrelationeditorwidget.h:258]: Please use the std C++ smart pointer instead of the qt one here

Comment (m-kuhn) [reply src/gui/qgsrelationeditorwidget.h:258]: Could this be a unique_ptr?

Comment (domi4484) [reply src/gui/qgsrelationeditorwidget.h]: @m-kuhn @nyalldawson oh I realise now that there was no memory leak, i did not expect to find the delete in [qgsdualview](https://github.com/qgis/QGIS/blob/50fd3d42593d7ef167982c5d1d64fe1bef46147d/src/gui/attributetable/qgsdualview.cpp#L1267).
There was still one leak when the dual view get deleted

File: src/gui/attributetable/qgsdualview.cpp [hunk 2]
```diff
@@ -116,6 +116,9 @@ QgsDualView::~QgsDualView()
 {
   QgsSettings settings;
   settings.setValue( QStringLiteral( "/qgis/attributeTable/splitterState" ), mConditionalSplitter->saveState() );
+
+  if ( mFeatureSelectionManager && mFeatureSelectionManager->parent() == this )
+    delete mFeatureSelectionManager;
```
Comment (m-kuhn) [inline src/gui/attributetable/qgsdualview.cpp:121]: Is this not already done by Qt (QObject parent/child ownership)?

Comment (domi4484) [reply src/gui/attributetable/qgsdualview.cpp:121]: Ehm yes. So I'll close this :)

**Pull request closed without merging**